tests/ui/async-await/in-trait/missing-feature-flag.rs

```diff
@@ -0,0 +1,25 @@
+// [next] compile-flags: -Zlower-impl-trait-in-trait-to-assoc-ty
```

Comment:
This needs edition 2021, and can have the AFIT feature gate enabled. The ICE is not dependent on either of those.